docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,418 @@
+ - `g` is aligned w.r.t `g.parent.foo`
+ - `g` is flipped w.r.t `g.flippedParent.foo`
+ - `g` is flipped w.r.t `g.parent.bar`
+
+Consider the following alignment relationships starting from `g.parent` and `g.flippedParent`. *Note that whether `g.parent` is aligned/flipped relative to `g` has no effect on the aligned/flipped relationship between `g.parent` and `g.parent.foo` because alignment is only relative to the two members in question!*:
+ - `g.parent` is aligned w.r.t. `g.parent.foo`
+ - `g.flippedParent` is aligned w.r.t. `g.flippedParent.foo`
+ - `g.parent` is flipped w.r.t. `g.parent.bar`
+ - `g.flippedParent` is flipped w.r.t. `g.flippedParent.bar`
+
+In summary, a field is aligned or flipped w.r.t. another member of the hardware component. This means that the type of the consumer/producer is the only information needed to determine the behavior of any operator. *Whether the consumer/producer is a subfield of a larger bundle is irrelevant; you ONLY need to know the type of the consumer/producer*.
+
+## Input/Output
+
+`Input(gen)`/`Output(gen)` are coercing operators. They perform two functions: (1) create a new Chisel type that has all flips removed from all recursive children fields but structurally equivalent to `gen`, and (2) apply `Flipped` if `Input`, keep aligned (do nothing) if `Output`. E.g. if we imagine a function called `cloneChiselTypeButStripAllFlips`, then `Input(gen)` is equivalent to `Flipped(cloneChiselTypeButStripAllFlips(gen))`.
+
+Note that if `gen` is a non-aggregate, then `Input(nonAggregateGen)` is equivalent to `Flipped(nonAggregateGen)`.
+
+> Future work will refactor how these primitives are exposed to the user to make Chisel's type system more intuitive.
+
+With this in mind, we can consider the following examples and detail relative alignments of fields.
+
+First, we can use a similar example to `Parent` but use `Input/Output` instead of `Flipped`.
+Because `foo` and `bar` are non-aggregates, `Input` is basically just a `Flipped` and thus the alignments are unchanged compared to the previous `Parent` example.
+
+```scala mdoc:silent
+import chisel3._
+class ParentWithOutputInput extends Bundle {
+  val foo = Output(UInt(32.W)) // Equivalent to just UInt(32.W)
+  val bar = Input(UInt(32.W))  // Equivalent to Flipped(UInt(32.W))
+}
+class MyModule2 extends Module {
+  val p = Wire(new ParentWithOutputInput)
+}
+```
+
+The aligments are the same as the previous `Parent` example:
+ - `p` is aligned w.r.t `p`
+ - `p.foo` is aligned w.r.t `p.foo`
+ - `p.bar` is aligned w.r.t `p.bar`
+ - `p` is aligned w.r.t `p.foo`
+ - `p` is flipped w.r.t `p.bar`
+ - `p.foo` is flipped w.r.t `p.bar`
+
+The next example has a nested bundle `GrandParent` who instantiates an `Output` `ParentWithOutputInput` field and an `Input` `ParentWithOutputInput` field.
+
+```scala mdoc:silent
+import chisel3._
+class GrandParent extends Bundle {
+  val o = Output(new ParentWithOutputInput())
+  val i = Input(new ParentWithOutputInput())
```

Comment:
```suggestion
  val coercedAligned = Output(new ParentWithOutputInput())
  val coercedFlipped = Input(new ParentWithOutputInput())
```

maybe? I like the o and i too, but keeping aligned and fliipped terminology could help?